### PR TITLE
Hide Hive ORC ACID DML behind feature flag (enabled by default)

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java [hunk 1]
```diff
@@ -146,6 +148,8 @@
 
     private boolean legacyHiveViewTranslation;
 
+    private boolean orcAcidDmlEnabled = true;
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java:151]: `orcAcidWritesEnabled`

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 2]
```diff
@@ -326,6 +326,7 @@
     private final HiveRedirectionsProvider hiveRedirectionsProvider;
     private final HiveMaterializedViewMetadata hiveMaterializedViewMetadata;
     private final AccessControlMetadata accessControlMetadata;
+    private final boolean orcAcidDmlEnabled;
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:329]: `orcAcidWritesEnabled`

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-kerberos-hdfs-impersonation-with-wire-encryption/hive.properties [hunk 3]
```diff
@@ -20,3 +20,4 @@ hive.security=sql-standard
 hive.translate-hive-views=true
 
 hive.hdfs.wire-encryption.enabled=true
+
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-kerberos-hdfs-impersonation-with-wire-encryption/hive.properties:23]: undo

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java [hunk 4]
```diff
@@ -1041,4 +1045,17 @@ public boolean isLegacyHiveViewTranslation()
     {
         return this.legacyHiveViewTranslation;
     }
+
+    @Config(ORC_ACID_WRITES_ENABLED_CONFIG_PROPERTY)
+    @ConfigDescription("Enable writes to ORC ACID transactional tables")
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java:1050]: Perhaps add a link to code-comment link to known issue(s)

**Review by findepi — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java (hunk 2 above)
Comment (losipiuk) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:329]: fixed (and other places too)

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java (hunk 4 above)
Comment (losipiuk) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java:1050]: added `//TODO` with issue link

**Review by findepi — APPROVED** [review]

**Review by electrum — CHANGES_REQUESTED** [review]
Let's hold off on this change. Changing the delete behavior for whole partitions (which was an explicit design choice, but I'm now convinced is the wrong way) should be an easy change.

**Comment by electrum** [comment]
Both #7291 and #7298 are resolved, so I think we can close this.

**Pull request closed without merging**